charts/backstage/templates/tests/test-connection.yaml

```diff
@@ -12,7 +12,26 @@ metadata:
 spec:
   containers:
     - name: curl
+      securityContext:
+        allowPrivilegeEscalation: false
+        readOnlyRootFilesystem: true
+        capabilities:
+          drop: ["ALL"]
+        runAsNonRoot: false
```

Comment:
This container runs as user `curl_user` which kubernetes can NOT verify is not a root user, and OpenShift supplies ranges of user IDs that containers can run as , not the standard 1000 or 10000 ranges that regular Kubernetes uses, so we need to set the `runAsNonRoot` to false, but since this is only run in a testing environment I think that the risk is very low for this being abused in some way.

Reply:
But depending on the pod security policy enforced in the namespace, this pod might not be accepted, no?
With this, I am getting the following warnings:
```
$ helm test my-rhdh-from-helm

W0614 15:11:47.281784 3252864 warnings.go:70] would violate PodSecurity "restricted:v1.24": runAsNonRoot != true (container "curl" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "curl" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
[...]
```

> since this is only run in a testing environment I think that the risk is very low for this being abused in some way.

To me, the goal of this test pod is to kinda smoke-test a deployed Helm Release with `helm test`. So IMO, it can be run in a production environment against a deployed Release, no?

Since this pod runs nothing more than a simple `curl` waiting for a response, would it make sense to replace the image with another more secure image that includes `curl` (or even `wget`)? Maybe there are some minimal ones from the Red Hat Ecosystem Catalog already..
btw, I can see an open issue for using numeric ID for `USER` in `quay.io/curl/curl` image, which should fix this issue on K8s/OpenShift I guess: https://github.com/curl/curl-container/issues/25

Reply:
AFAIK this would only be used in our automated testing. If it was meant to be used in production I would think it would be some kind of a sidecar instead?

Reply:
@rm3l Did you have any other thoughts on this?

Reply:
Sounds good - thanks!